checker/match.go

```diff
@@ -174,3 +177,47 @@ func matchHTTPMethods(methods []string, reqMethod string) bool {
 	log.Debug("HTTP Method not matched.")
 	return false
 }
+
+func matchL4Protocol(rule *proto.Rule, dest *authz.AttributeContext_Peer) bool {
+	// Extract L4 protocol type of socket address for destination peer context. Match against rules.
+	if dest == nil {
+		log.Warn("nil request destination peer.")
+		return false
+	}
+
+	// default protocol is TCP
+	reqProtocol := "TCP"
+	if a := dest.GetAddress(); a != nil {
+		if sa := a.GetSocketAddress(); sa != nil {
+			reqProtocol = sa.GetProtocol().String()
+		}
+	}
```

Comment:
You should probably just convert `reqProtocol` to lowercase here, since you always compare it lowercase.

Reply:
Yes. Fixed.